src/main/java/com/google/step/datamanager/UserManager.java

```diff
@@ -0,0 +1,51 @@
+package com.google.step.datamanager;
+
+import com.google.step.model.User;
+import java.util.Optional;
+
+/** A UserManager handling user datastore operations. */
+public interface UserManager {
+
+  /**
+   * Returns a User object when the user with the email logs in.
+   *
+   * @param email email of the user who logs in.
+   * @return User object with the email.
+   */
+  public User userLogin(String email);
+
+  /**
+   * Creates a User object with the given email.
+   *
+   * @param email email of the user.
+   * @return User object with the email.
+   */
+  public User createUser(String email);
+
+  /**
+   * Returns a User object with the given id.
+   *
+   * @param id id of the user.
+   * @return User object with the id.
+   */
+  public User readDeal(long id);
+
+  /**
+   * Updates a user's info with the given id.
+   *
+   * @param id id of the user.
+   * @param email email of the user.
+   * @param username username of the user.
+   * @param photoBlobKey blob key of the profile photo if it exists.
+   * @param bio bio of the user.
+   */
+  public void updateUser(
+      long id, String email, String username, Optional<String> photoBlobKey, String bio);
```

Comment:
What if I want to update only one of these fields?

Reply:
I think update operation here is similar to overwriting the user object so any number of fields can be updated in one update call.

Reply:
But that means the caller needs to read the old object first to be able to pass the rest of the fields. I would rather suggest passing a partially populated user object as the argument and then only updating the fields that are populated. Plus in the future the User class could have many more fields, including all those fields as arguments here makes it too verbose.

Reply:
I see. I updated user manager. is it ok to set User property to be null?